common/unsafe/src/main/java/org/apache/spark/unsafe/array/LongArray.java

```diff
@@ -39,7 +39,7 @@
   private final long length;
 
   public LongArray(MemoryBlock memory) {
-    assert memory.size() < (long) Integer.MAX_VALUE * 8: "Array size > 4 billion elements";
+    assert memory.size() < (long) Integer.MAX_VALUE * 8: "Array size > 2.1 billion elements";
```

Comment:
`assert memory.size() <= (long) (Integer.MAX_VALUE - 8) * 8`

Reply:
ISTM `memory.size() < (long) Integer.MAX_VALUE * 8` is correct? see: https://github.com/apache/spark/blob/c66d64b3df9d9ffba0b16a62015680f6f876fc68/common/unsafe/src/main/java/org/apache/spark/unsafe/memory/MemoryBlock.java#L54

Reply:
We need the same assert below?
 https://github.com/apache/spark/blob/c66d64b3df9d9ffba0b16a62015680f6f876fc68/common/unsafe/src/main/java/org/apache/spark/unsafe/memory/MemoryBlock.java#L53

Reply:
maybe also add the exact number in the error message instead of `2.1 billion`